src/payment/strategies/stripev3/stripev3-payment-strategy.spec.ts

```diff
@@ -0,0 +1,260 @@
+        const stripeV3JsMock = getStripeV3JsMock();
+
+        beforeEach(() => {
+            stripeV3Options = getStripeV3InitializeOptionsMock();
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve(store.getState()));
+            jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(getStripeV3());
+            jest.spyOn(orderActionCreator, 'submitOrder').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentActionCreator, 'submitPayment').mockReturnValue(Promise.resolve());
+        });
+
+        it('creates the order and submit payment', async () => {
+            stripeV3JsMock.handleCardPayment = jest.fn(
+                () => Promise.resolve(getStripeV3HandleCardResponse())
+            );
+
+            jest.spyOn(stripeScriptLoader, 'load').mockReturnValue(Promise.resolve(stripeV3JsMock));
+
+            await strategy.initialize(stripeV3Options);
+            const response = await strategy.execute(getStripeV3OrderRequestBodyMock());
+
+            expect(store.getState().paymentMethods.getPaymentMethod).toHaveBeenCalled();
+            expect(stripeV3JsMock.handleCardPayment).toHaveBeenCalled();
+            expect(orderActionCreator.submitOrder).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalled();
+            expect(response).toBe(store.getState());
+        });
+
+        it('throws an error when payment is not set properly into payload', async () => {
+            const payload = {
+                payment: undefined,
+            };
+
+            expect(() => strategy.execute(payload))
+                .toThrow(PaymentArgumentInvalidError);
+        });
+
+        it('throws an error when store dispatch does not load paymentMethod', async () => {
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.reject({message: 'error'}));
+
+            const response = strategy.execute(getStripeV3OrderRequestBodyMock());
+
+            expect(response).rejects.toThrowError('error');
+            expect(response).rejects.toBeInstanceOf(StandardError);
+        });
+
+        it('throws an error when getPaymentMethod will not retrieve clientToken', async () => {
+            const stripeV3 = {
+                id: 'stripev3',
+                logoUrl: '',
+                method: 'stripev3',
+                supportedCards: [],
+                config: {
+                    displayName: 'Stripe',
+                    merchantId: '',
+                    testMode: true,
+                },
+                type: 'PAYMENT_TYPE_API',
+            };
+
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(stripeV3);
+
+            const response = strategy.execute(getStripeV3OrderRequestBodyMock());
+
+            expect(response).rejects.toBeInstanceOf(StandardError);
```

Comment:
If you want to use `rejects` or `resolves`, you'll need to add `return` before the `expect`. i.e.:

```ts
return expect(response).rejects.toBeInstanceOf(StandardError);
```

Otherwise, the test won't wait for the promise to reject / resolve and just pass itself.